app/services/courses/query.rb

```diff
@@ -13,16 +13,34 @@ def self.call(...)
     def initialize(params:)
       @params = params
       @applied_scopes = {}
-      @scope = RecruitmentCycle
-               .current
-               .courses
-               .joins(:site_statuses)
-               .where(
-                 site_statuses: {
-                   status: SiteStatus.statuses[:running],
-                   publish: SiteStatus.publishes[:published],
-                 },
-               )
+      @current_recruitment_cycle = RecruitmentCycle.current
+
+      @scope = Course
+                 .joins(
+                   <<~SQL,
+                     INNER JOIN (
+                       SELECT id, provider_name, provider_code, recruitment_cycle_id
+                       FROM provider
+                       WHERE recruitment_cycle_id = #{@current_recruitment_cycle.id}
+                         AND discarded_at IS NULL
+                     ) provider ON course.provider_id = provider.id
+                     INNER JOIN course_site site_statuses ON (
+                       site_statuses.course_id = course.id
+                       AND site_statuses.status = 'R'
+                       AND site_statuses.publish = 'Y'
+                     )
+                   SQL
+                 )
+                 .where(course: { discarded_at: nil })
+                 .where(
+                   provider: { recruitment_cycle_id: @current_recruitment_cycle.id },
+                 )
+                 .where(
+                   site_statuses: {
+                     status: SiteStatus.statuses[:running],
+                     publish: SiteStatus.publishes[:published],
+                   },
+                 )
```

Comment:
this is not needed and postgres will ignore.

BUT somethings (gut feeling!) tells me if someone remove the 
`WHERE recruitment_cycle_id =  #{@current_recruitment_cycle.id}` in the JOIN will be a disaster (the specs will fail but either way). So I decided to keep. No diff, postgres will ignore.
```
.where(
                   provider: { recruitment_cycle_id: @current_recruitment_cycle.id },
                 )
                 .where(
                   site_statuses: {
                     status: SiteStatus.statuses[:running],
                     publish: SiteStatus.publishes[:published],
                   },
                 )
```